### PR TITLE
feature/1437 extract nest method

### PR DESCRIPTION
**Pull request by DEMONSTAT**
This is my second time to create the pull request. Last PR is closed by me because some problem.

**Describe the refactoring action**
An example of the issue：
In the part of the main package I modified, I found that the content of the has method is the same, so I tried to extract this part and then package it, so that the packaged code part becomes less difficult to understand.

**Expected benefit**
Long code segments are not conducive to understanding and modification. For these large pieces of code, we can refactor them to improve readability and modifiability. I don't think it's very difficult to make specific changes, but I spent a lot of time understanding the code. After my modifications, the code became more orderly. Good design is half the battle. Stopping to improve the design through refactoring may slow down the current speed, but the latecomer advantage it brings cannot be underestimated.

### CODE REVIEW

File: inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/querybuilder/SPARQLQueryBuilder.java [hunk 1]
```diff
@@ -450,7 +426,23 @@ protected GraphPattern rootsPattern(KnowledgeBase aKb)
             }            
         }
     }
-    
+
+    private static void selectValue(KnowledgeBase aKB, Iri aContext, List<GraphPattern> classPatterns) {
```
Comment (munterkalmsteiner) [inline inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/querybuilder/SPARQLQueryBuilder.java]: I think you should refactor these two methods into one. Maybe you could have a method like `private void createPattern(KnowledgeBase aKB, RdfSubject subject, RdfObject object, List<GraphPattern> patterns)`. When called, subject and object can be defined as needed.

File: inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/querybuilder/SPARQLQueryBuilderGenericTest.java [hunk 2]
```diff
@@ -127,29 +127,37 @@ public void setup() throws Exception
 
         switch (kb.getType()) {
         case LOCAL: {
-            LuceneSail lucenesail = new LuceneSail();
-            lucenesail.setParameter(LuceneSail.LUCENE_RAMDIR_KEY, "true");
-            lucenesail.setBaseSail(new MemoryStore());
-            repo = new SailRepository(lucenesail);
-            repo.init();
-            importData(repo, profile.getAccess().getAccessUrl());
+            localRepo();
```
Comment (munterkalmsteiner) [inline inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/querybuilder/SPARQLQueryBuilderGenericTest.java]: Maybe `createLocalRepo` is better as it indicates the action.

File: inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/querybuilder/SPARQLQueryBuilderGenericTest.java [hunk 3]
```diff
@@ -127,29 +127,37 @@ public void setup() throws Exception
...
         }
         case REMOTE: {
-            Assume.assumeTrue(
-                    "Remote repository at [" + profile.getAccess().getAccessUrl()
-                            + "] is not reachable",
-                    isReachable(profile.getAccess().getAccessUrl()));
-            
-            repo = new SPARQLRepository(profile.getAccess().getAccessUrl());
-            repo.init();
+            RemoteRepo();
```
Comment (munterkalmsteiner) [inline inception-kb/src/test/java/de/tudarmstadt/ukp/inception/kb/querybuilder/SPARQLQueryBuilderGenericTest.java]: Same here. `createRemoteRepo()`

**Comment by munterkalmsteiner** [comment]
@DEMONSTAT do you intend to work on my suggested changes?

**Comment by reckart** [comment]
Assuming this is no longer going to be worked on.

**Pull request closed without merging**